src/components/shared/ButtonWithPanel.js

```diff
@@ -4,38 +4,40 @@
 
 // @flow
 
-import React, { PureComponent } from 'react';
+import * as React from 'react';
 import classNames from 'classnames';
 
 require('./ButtonWithPanel.css');
 
+type PanelProps = {
+  onOpen?: () => mixed,
+  onClose?: () => mixed,
+};
+
 interface Panel {
-  props: {
-    onOpen: () => mixed,
-    onClose: () => mixed,
-  },
   open(): mixed,
 }
 
 type Props = {
   className: string,
   label: string,
-  panel: React$Element<*>, // Ideally we'd like to say that panel implements Panel, but I can't express it with Flow
+  panel: React.Element<
+    Class<Panel & React.Component<$Subtype<PanelProps>, any>>
+  >,
```

Comment:
yay, better typing :)

Reply:
Holy cow, that's a type. Should that last `any` be mixed? Or does it matter?

Reply:
I'd say it doesn't matter because we don't use its state here.

Reply:
```
// Almost all of the time when a user writes `React.Element<*>` they meant
// `React.Element<any>`. They just felt bad about writing `any`.
```
from https://github.com/facebook/flow/blob/c803f7abe609a896a0f54e3654e99348a9f50370/packages/flow-upgrade/src/upgrades/0.53.0/ReactUtilityTypes/codemod.js#L162-L163